### PR TITLE
rgw/lua: Example read/write of StorageClass field

### PR DESCRIPTION
**Pull request by curtbruns**
Admins may setup different pools for RGW objects and
having the StorageClass field mutable allows the steering
of RGW objects to the proper pools.  This Lua example shows
how a user can modify the StorageClass header when
it is empty on an PUT request.

Signed-off-by: Curt Bruns <curt.e.bruns@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by curtbruns** [comment]
@yuvalif @cbodley Hi Casey and Yuval - here is the PR that provides the example and documentation of the StorageClass modification that I demoed in the RGW Refactor meeting.   Thanks!

**Review by yuvalif — APPROVED** [review]
please add size based steering to the example

File: examples/lua/storage_class.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Introduction
+
+This directory contains an example `storage_class.lua` on how to
```
[the remark this answers is not in the document]
Comment (curtbruns) [reply examples/lua/storage_class.md:3]: @Yuval - No problem.  I've updated the example to show that an an object >= 64KiB gets it's storage class set to QLC_CLASS, and any other object is modified to STANDARD (default) class.  This only happens if there isn't a storage class already present in the PUT request.

**Comment by yuvalif** [comment]
jenkins test make check

**Pull request merged**